synapse/utils/stormcov/plugin.py

```diff
@@ -243,4 +243,39 @@ def lines(self):
             if token.type in TOKENS:
                 source_lines.add(token.line)
 
-        return source_lines
+        return source_lines - self.excluded_lines()
+
+    def excluded_lines(self):
+        excluded_lines = set()
+
+        pragma = 'pragma: no cover'
+        start = 'pragma: no cover start'
+        stop = 'pragma: no cover stop'
+
+        lines = self.source().splitlines()
+        nocov = [(lineno + 1, text) for (lineno, text) in enumerate(lines) if pragma in text]
+
+        block = None
+        for (lineno, text) in nocov:
+            # Invalid condition
+            if stop in text and block is None:
+                continue
+
+            # Invalid condition
+            if start in text and block is not None:
+                continue
+
+            if start in text:
```

Comment:
I think you could merge these with the previous blocks to avoid checking for start/stop in the line twice